packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -282,6 +283,25 @@ module AlgebraicCombination = {
       }
     }
   }
+
+  let run = (
+    ~mode: option<asMode>=?,
+    t1: t,
+    ~toPointSetFn: toPointSetFn,
+    ~toSampleSetFn: toSampleSetFn,
+    ~arithmeticOperation,
+    ~t2: t,
+  ): result<t, error> => {
+    let algebraicResult = run'(t1, ~toPointSetFn, ~toSampleSetFn, ~arithmeticOperation, ~t2)
```

Comment:
It should be overriding the way this is done. You need to change the ``run`` functionality for this.